src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -243,13 +243,30 @@ impl<'a, 'tcx> Qualifier<'a, 'tcx, 'tcx> {
             return;
         }
 
+        if self.tcx.features().const_let {
+            let mut dest = dest;
+            let index = loop {
+                match dest {
+                    Place::Local(index) => break *index,
+                    Place::Projection(proj) => dest = &proj.base,
+                    Place::Promoted(..) => bug!("promoteds don't exist yet during promotion"),
+                    Place::Static(..) => {
+                        // Catch more errors in the destination.
+                        self.visit_place(
+                            dest,
+                            PlaceContext::MutatingUse(MutatingUseContext::Store),
+                            location
+                        );
```

Comment:
The old branch also calls `self.statement_like();`. This one doesn't. What does that do?

Reply:
`statement_like` emits an error in const contexts and marks it as unpromotable otherwise.

...
...
...

I guess I should add some tests to ensure that we don't promote `let x: &'static u32 = { let y = 42; &y };` or some other nonsense. And generally look at this code closer.

Reply:
ah nevermind, non-const contexts (so promotion) is handled above and an early abort makes sure we never even reach this code.

So, for statics we don't need to run `statement_like`, because `visit_place` correctly checks that only statics can access other statics.

Reply:
Okay, not sure I fully understand but I do see the `if self.mode == Mode::Fn` above.

Would you add some comments explaining this?^^